tools/renovate-post-upgrade.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# This script is run after Renovate upgrades dependencies or lock files.
+
+set -euo pipefail
+
+# Download and install cargo-hakari if it is not already installed.
+if ! command -v cargo-hakari &> /dev/null; then
+    # Need cargo-binstall to install cargo-hakari.
+    if ! command -v cargo-binstall &> /dev/null; then
+        # Fetch cargo binstall.
+        echo "Installing cargo-binstall..."
+        curl -L --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/cargo-bins/cargo-binstall/main/install-from-binstall-release.sh | bash
+    fi
+
+    # Install cargo-hakari.
+    echo "Installing cargo-hakari..."
+    cargo binstall cargo-hakari --no-confirm
```

Comment:
@davepacheco mentioned that "we should expect all our cloud services to fail transiently at any time", in https://github.com/oxidecomputer/omicron/issues/4165 . Do you think we should build a retry-loop here, for these installation commands?

Reply:
Huh, for some reason I thought binstall already did retries but it turns out it does not (https://github.com/cargo-bins/cargo-binstall/pull/131 was rejected). Yes, retries are worth adding here for sure.

Reply:
Added retries.